tests/test_buffered_mode.py

```diff
@@ -38,6 +38,13 @@
         os.chmod(path, mode)
 
 
+def deprecated_call(fun):
```

Comment:
While I'm generally a fan of tricks like this, in this case I'd prefer to use the `pytest.deprecated_call` CM explicitly at the exact location where it makes sense. The reason is that marking the complete unit test like this will make it difficult to follow which specific function is expected to trigger an expectation warning.

Reply:
done that! please review if this was required. Seeing the repetitive work the lazy programmer in me motivated me to take the decorator path😅 . But the reason that you made completely makes sense.